libraries/lib-preferences/FileConfig.h

```diff
@@ -12,39 +12,39 @@
 #define __AUDACITY_WIDGETS_FILECONFIG__
 
 #include <memory>
-
-#include <wx/defs.h>
-#include <wx/fileconf.h>
-#include "Identifier.h"
-
-class PREFERENCES_API FileConfig : public wxConfigBase
+#include <wx/confbase.h>
+
+///@brief Wraps wxConfigBase adding dirty flag that prevents extensive
+///redundant writes when wxConfigBase::Flush() is called. Replicates
+///wxConfigBase interface.
+///Requires another wxConfigBase instance that is used as an actual
+///storage (e.g. all write/reads are delegated to that object)
+//TODO: wxFileConfig already has dirty flag
+class PREFERENCES_API FileConfig final : public wxConfigBase
```

Comment:
This looks like a proper hotfix, but I would instead push it further. 

1. I would like it to use Audacity specific interface. Much smaller one, only providing the functionality Audacity needs. `SetPath` is redundant as well.
2. I don't quite get the need for `SetImpl`, especially because it expects to implement the same interface. I think something like `GlobalHook` should be used to initialize `gPrefs` (or anything else really).

Reply:
> I don't quite get the need for SetImpl, especially because it expects to implement the same interface

It used to create `wxFileConfig` inside `FileConfig` which was performing all the actual job. Now it injected instead. This refactoring made it clear that the only purpose of `FileConfig` is prevent redundant disk writes, which is actually what `wxFileConfig` do as well...